library/ecdsa.c

```diff
@@ -540,7 +540,7 @@ static int ecdsa_verify_restartable( mbedtls_ecp_group *grp,
     if( mbedtls_mpi_cmp_int( r, 1 ) < 0 || mbedtls_mpi_cmp_mpi( r, &grp->N ) >= 0 ||
         mbedtls_mpi_cmp_int( s, 1 ) < 0 || mbedtls_mpi_cmp_mpi( s, &grp->N ) >= 0 )
     {
-        ret = MBEDTLS_ERR_ECP_VERIFY_FAILED;
+        ret = MBEDTLS_ERR_ECP_BAD_INPUT_DATA;
```

Comment:
Good spot. From looking at the code, it seems to change from `PSA_ERROR_INVALID_SIGNATURE` (according to the PSA spec, this means "signature is not the result of signing hash with algorithm alg using the private key corresponding to key.") to `PSA_ERROR_INVALID_ARGUMENT` (PSA spec: "key is not a public key or an asymmetric key pair, that is compatible with alg"). So that change is an improvement I think.

I will add a test.

Reply:
Here PSA is explicit that you don't get `INVALID_ARGUMENT` from a bad signature, only from a bad public key. So if the low-level code returns the same error code for both, it's going to be hard to get the PSA error code right.

Reply:
Ah right, I mis-read this as applying to bad signatures as well. I'm going to drop the change to the error code as it doesn't seem to be helpful. I'll also add a correction to the documentation for `mbedtls_ecdsa_verify` to reflect its actual behaviour.